src/o365/spo/commands/folder/folder-move.ts

```diff
@@ -0,0 +1,340 @@
+import auth from '../../SpoAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import * as request from 'request-promise-native';
+import {
+  CommandOption,
+  CommandValidate
+} from '../../../../Command';
+import SpoCommand from '../../SpoCommand';
+import Utils from '../../../../Utils';
+import { Auth } from '../../../../Auth';
+import * as url from 'url';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  sourceUrl: string;
+  targetUrl: string;
+  ignoreVersionHistory: boolean;
+  allowSchemaMismatch: boolean;
+}
+
+interface JobProgressOptions {
+  webUrl: string;
+  accessToken: string;
+  /**
+   * Response object retrieved from /_api/site/CreateCopyJobs
+   */
+  copyJopInfo: any;
+  /**
+   * Poll interval to call /_api/site/GetCopyJobProgress
+   */
+  progressPollInterval: number;
+  /**
+   * Max poll intervals to call /_api/site/GetCopyJobProgress
+   * after which to give up
+   */
+  progressMaxPollAttempts: number;
+  /**
+   * Retry attempts before give up.
+   * Give up if /_api/site/GetCopyJobProgress returns 
+   * X reject promises in a row
+   */
+  progressRetryAttempts: number;
+}
+
+class SpoFolderMoveCommand extends SpoCommand {
+  public get name(): string {
+    return commands.FOLDER_MOVE;
+  }
+
+  public get description(): string {
+    return 'Moves a folder to another location';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const resource: string = Auth.getResourceFromUrl(args.options.webUrl);
+    let siteAccessToken: string = '';
+    const webUrl: string = args.options.webUrl;
+    const tenantUrl: string = `${url.parse(webUrl).protocol}//${url.parse(webUrl).hostname}`;
+
+    if (this.debug) {
+      cmd.log(`Retrieving access token for ${resource}...`);
+    }
+
+    auth
+      .getAccessToken(resource, auth.service.refreshToken as string, cmd, this.debug)
+      .then((accessToken: string): request.RequestPromise => {
+        if (this.debug) {
+          cmd.log(`Retrieved access token ${accessToken}.`);
+        }
+
+        siteAccessToken = accessToken;
+
+        const sourceAbsoluteUrl: string = this.urlCombine(webUrl, args.options.sourceUrl);
+        const ignoreVersionHistory: boolean = args.options.ignoreVersionHistory || true;
```

Comment:
If the default is `true` and specifying the option is also `true`, how would you set `false`?

Reply:
Hmm, you're right ofcourse. @waldekmastykarz how to you propose I fix this?
Should the default be false, or how would I go about this?

Reply:
If the default value is `true`, then name of the switch should be the opposite, like `--includeVersionHistory` which would translate to a `false` value in the `IgnoreVersionHistory` option in the request payload. It is possible to negate switches, using `--no-ignoreVersionHistory` but it's pretty obscure and not many people know about it, so I'd be in favor of renaming the switch as I mentioned first.

Reply:
You live you learn, thanks :)